packages/url/src/index.js

```diff
@@ -0,0 +1,127 @@
+import crypto from 'crypto';
+import path from 'path';
+import fs from 'fs';
+
+import makeDir from 'make-dir';
+import mime from 'mime';
+import { createFilter } from 'rollup-pluginutils';
+
+const defaultInclude = ['**/*.svg', '**/*.png', '**/*.jpg', '**/*.gif'];
+
+export default function url(options = {}) {
+  const {
+    limit = 14 * 1024,
+    include = defaultInclude,
+    exclude,
+    publicPath = '',
+    emitFiles = true,
+    fileName = '[hash][extname]'
+  } = options;
+  const filter = createFilter(include, exclude);
+
+  const copies = Object.create(null);
+
+  return {
+    load(id) {
+      if (!filter(id)) {
+        return null;
+      }
+      return Promise.all([promise(fs.stat, id), promise(fs.readFile, id)]).then(
+        ([stats, buffer]) => {
+          let data;
+          if ((limit && stats.size > limit) || limit === 0) {
+            const hash = crypto
+              .createHash('sha1')
+              .update(buffer)
+              .digest('hex')
+              .substr(0, 16);
+            const ext = path.extname(id);
+            const name = path.basename(id, ext);
+            // Determine the directory name of the file based
+            // on either the relative path provided in options,
+            // or the parent directory
+            const relativeDir = options.sourceDir
+              ? path.relative(options.sourceDir, path.dirname(id))
+              : path
+                  .dirname(id)
+                  .split(path.sep)
+                  .pop();
+
+            // Generate the output file name based on some string
+            // replacement parameters
+            const outputFileName = fileName
+              .replace(/\[hash\]/g, hash)
+              .replace(/\[extname\]/g, ext)
+              .replace(/\[dirname\]/g, `${relativeDir}/`)
+              .replace(/\[name\]/g, name);
+            data = `${publicPath}${outputFileName}`;
+            copies[id] = outputFileName;
+          } else {
+            const mimetype = mime.getType(id);
+            const isSVG = mimetype === 'image/svg+xml';
+            data = isSVG ? encodeSVG(buffer) : buffer.toString('base64');
+            const encoding = isSVG ? '' : ';base64';
+            data = `data:${mimetype}${encoding},${data}`;
+          }
+          return `export default "${data}"`;
+        }
+      );
+    },
+    generateBundle: async function write(outputOptions) {
+      // Allow skipping saving files for server side builds.
+      if (!emitFiles) return;
+
+      const base = options.destDir || outputOptions.dir || path.dirname(outputOptions.file);
+
+      await makeDir(base);
+
+      Promise.all(
```

Comment:
if you are calling Promise.all here I assume you should await this 😉

Reply:
Nice catch, updated!